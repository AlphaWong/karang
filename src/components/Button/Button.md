#### Usage:

```js static
import { Buton } from 'lalamove-ui';
```

**Variant**
* `default` — this is the default.
* `link`
* `primary`
* `secondary`
* `danger`
* `outline`
* `dangerOutline`

```jsx static
<Button variant="primary">Text</Button>
```

```
<div style={{ display: 'flex', justifyContent: 'space-around' }}>
    <Button variant="link">Link</Button>
    <Button>Default</Button>
    <Button variant="primary">Primary</Button>
    <Button variant="secondary">Secondary</Button>
    <Button variant="danger">Danger</Button>
    <Button variant="outline">Outline</Button>
    <Button variant="dangerOutline">Outline</Button>
</div>
```

**Size**
* `default` — this is the default.
* `small`
* `large`
* `xlarge`

```jsx static
<Button size="large">Text</Button>
```

```
<div style={{ display: 'flex', alignItems: 'center', justifyContent: 'space-around' }}>
    <Button variant="primary" size="small">Small</Button>
    <Button variant="primary">Default</Button>
    <Button variant="primary" size="large">Large</Button>
    <Button variant="primary" size="xlarge">Extra Large</Button>
</div>
```

**Block**

Passing `block` as true to make the button fit the width to its parent width.

```jsx static
<Button block>Text</Button>
```

```
<Button variant="primary" size="small" block>Small</Button>
<br />
<Button variant="primary" block>Default</Button>
<br />
<Button variant="primary" size="large" block>Large</Button>
<br />
<Button variant="primary" size="xlarge" block>Extra Large</Button>
```

**Button with icon**

It supports to pass other components after the text, and it is usually used with `Icon`. Please avoid using two icons in the button.

```jsx static
<Button icon={<Icon type="onOff" />}>Logout</Button>
```

```
<Button variant="dangerOutline" icon={<Icon type="onOff" />}>Logout</Button>
```

**Loading**

Show spinner when it is `true`. This also disables the button.

```jsx static
<Button variant="primary" isLoading>Text</Button>
```

```
<div style={{ display: 'flex', justifyContent: 'space-around' }}>
    <Button variant="link" isLoading>Link</Button>
    <Button isLoading>Default</Button>
    <Button variant="primary" isLoading>Primary</Button>
    <Button variant="secondary" isLoading>Secondary</Button>
    <Button variant="danger" isLoading>Danger</Button>
    <Button variant="outline" isLoading>Outline</Button>
    <Button variant="dangerOutline" isLoading>Outline</Button>
</div>
```
