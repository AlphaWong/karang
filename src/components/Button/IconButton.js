import React from 'react';
import { node } from 'prop-types';
import styled from 'styled-components';
import { orange, offWhite } from 'styles/colors';

const Button = styled.button`
  appearance: none;
  padding: 5px;
  border: none;
  color: ${orange};
  cursor: pointer;
  border-radius: 2px;
  outline: 0;

  &:hover,
  &:focus {
    background: ${offWhite};
  }
`;

const IconButton = props => <Button {...props}>{props.children}</Button>;

IconButton.propTypes = {
  children: node.isRequired, // eslint-disable-line react/no-typos
};

export default IconButton;
