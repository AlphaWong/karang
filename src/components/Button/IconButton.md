#### Usage:

```js static
import { IconButon } from 'lalamove-ui';
```

**Variant**
* `default` — this is the default.
* `primary`
* `secondary`
* `danger`

```jsx static
<IconButton variant="primary">
    <Icon ... />
</IconButton>
```

```
<div style={{ display: 'flex', justifyContent: 'space-around' }}>
    <IconButton>
      <Icon type="alert" size={24} />
    </IconButton>
    <IconButton variant="primary">
      <Icon type="alert" size={24} />
    </IconButton>
    <IconButton variant="secondary">
      <Icon type="alert" size={24} />
    </IconButton>
    <IconButton variant="danger">
      <Icon type="alert" size={24} />
    </IconButton>
</div>
```

**Shape**
* `default` — this is the default.
* `classic`

```jsx static
<IconButton shape="classic">
    <Icon ... />
</IconButton>
```

```
<div style={{ display: 'flex', justifyContent: 'space-around' }}>
    <IconButton>
      <Icon type="close" size={24} />
    </IconButton>
    <IconButton shape="classic">
      <Icon type="close" size={24} />
    </IconButton>
</div>
```
