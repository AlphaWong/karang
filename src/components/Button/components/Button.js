import React from 'react';
import { bool, oneOf, node } from 'prop-types';
import styled, { css } from 'styled-components';
import { transparentize, darken } from 'polished';

import Base from './style';
import omit from 'utils/omit';
import colors from 'styles/colors';
import loaderGif from 'assets/loader.gif';

/* eslint-disable import/no-named-as-default-member */
const {
  orange: primaryColor,
  gold: secondaryColor,
  offWhite: lightGray,
  gray: darkGray,
} = colors;
/* eslint-enable import/no-named-as-default-member */

const Button = styled(Base)`
  /* layout */
  display: ${({ block }) => (block ? 'block' : 'inline-block')};
  ${({ block }) => block && 'width: 100%;'};

  /* size */
  ${({ size }) => {
    switch (size) {
      case 'large':
        return css`
          padding: 0.75em 1em;
          font-size: 16px;
          font-weight: 700;
        `;
      case 'xlarge':
        return css`
          padding: 0.9em 1em;
          font-size: 18px;
          font-weight: 700;
        `;
      default:
        return css``;
    }
  }}

  /* type */
  ${({ variant }) => {
    switch (variant) {
      case 'primary':
        return css`
          background-color: ${primaryColor};
          border-color: ${primaryColor};
          color: white;
        `;
      case 'secondary':
        return css`
          background-color: ${secondaryColor};
          border-color: ${secondaryColor};
          color: white;
        `;
      case 'outline':
        return css`
          color: ${primaryColor};
          border-color: ${primaryColor};
        `;
      case 'link':
        return css`
          border: none;
          padding: 0;
          color: ${primaryColor};

          &:active {
            box-shadow: none;
            color: ${darken(0.2, primaryColor)};
            text-decoration: none !important;
          }

          &:hover:enabled,
          &:focus:enabled {
            box-shadow: none;
            color: ${transparentize(0.3, primaryColor)};
            text-decoration: underline;
          }
        `;
      default:
        return css`
          border-color: ${lightGray};
          color: ${darkGray};
        `;
    }
  }}

  /* loading */
  ${({ isLoading }) =>
    isLoading &&
    css`
      background-image: url(${loaderGif});
      background-position: center;
      background-size: 16px;
      background-repeat: no-repeat;
      color: rgba(0, 0, 0, 0);
    `}
`;

Button.defaultProps = {
  isLoading: false,
  size: null,
  variant: 'default',
  block: false,
  type: 'button',
};

Button.propTypes = {
  isLoading: bool,
  size: oneOf(['large', 'xlarge']),
  variant: oneOf(['default', 'primary', 'secondary', 'outline', 'link']),
  block: bool,
  children: node.isRequired,
  type: oneOf(['button', 'submit', 'reset']),
};

export function withComponent(Comp) {
  return Button.withComponent(props => {
    const propsToFilter = Object.keys(Button.propTypes);
    const filteredProps = omit(props, propsToFilter);
    return <Comp {...filteredProps}>{props.children}</Comp>;
  });
}

const Text = styled.span`
  vertical-align: middle;
  margin-right: 10px;
`;

// eslint-disable-next-line react/prop-types
export default ({ icon, children, ...rest }) => {
  if (icon) {
    return (
      <Button {...rest}>
        <Text>{children}</Text>
        {icon}
      </Button>
    );
  }
  return <Button {...rest}>{children}</Button>;
};
