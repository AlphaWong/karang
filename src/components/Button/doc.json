{
  "name": "Button",
  "import": "import { Button } from 'lalamove-ui';"
}
