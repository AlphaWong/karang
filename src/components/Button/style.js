import styled from 'styled-components';

import colors from 'styles/colors';
import { baseFontSize } from 'styles/scales';
import { primaryFonts } from 'styles/fonts';

const fontFamily = primaryFonts;
const lightGray = colors.offWhite; // eslint-disable-line

const Base = styled.button`
  /* structure */
  appearance: none;
  display: block;
  padding: 0.5em 1em;

  /* style */
  border: 1px currentColor solid;
  background-color: transparent;
  border-radius: 2px;
  font-family: ${fontFamily};
  font-size: ${baseFontSize}px;
  font-weight: 400;
  line-height: 1;
  cursor: pointer;
  outline: 0;

  &:disabled {
    opacity: 0.3;
    cursor: not-allowed;
  }

  &:hover:enabled,
  &:focus:enabled {
    box-shadow: 0 2px 4px rgba(0, 0, 0, 0.22);
  }

  &:active:enabled {
    box-shadow: none;
  }
`;

export default Base;
