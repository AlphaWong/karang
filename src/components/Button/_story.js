import React from 'react';
import styled from 'styled-components';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import Icon from 'components/Icon';
import Button, { ButtonWithComponent, CloseButton } from './index';
import { white, primary } from 'styles/colors';

const Container = styled.div`
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
  align-items: center;
`;

const CustomComponent = ButtonWithComponent('a');

storiesOf('Button', module).add(
  'Basic',
  withInfo({})(() => (
    <div>
      <h4>Types</h4>
      <Container>
        <Button variant="link">Link</Button>
        <Button>Default</Button>
        <Button variant="primary">Primary</Button>
        <Button variant="secondary">Secondary</Button>
        <Button variant="outline">Outline</Button>
      </Container>
      <h4>Large</h4>
      <Container>
        <Button variant="link" size="large">
          Link
        </Button>
        <Button size="large">Default</Button>
        <Button variant="primary" size="large">
          Primary
        </Button>
        <Button variant="secondary" size="large">
          Secondary
        </Button>
        <Button variant="outline" size="large">
          Outline
        </Button>
      </Container>
      <h4>Extra Large</h4>
      <Container>
        <Button variant="link" size="xlarge">
          Link
        </Button>
        <Button size="xlarge">Default</Button>
        <Button variant="primary" size="xlarge">
          Primary
        </Button>
        <Button variant="secondary" size="xlarge">
          Secondary
        </Button>
        <Button variant="outline" size="xlarge">
          Outline
        </Button>
      </Container>
      <h4>Block</h4>
      <Button block variant="primary">
        block primary
      </Button>
      <br />
      <Button block>block</Button>
      <h4>Loading</h4>
      <Container>
        <Button isLoading>Demo</Button>
        <Button variant="primary" isLoading>
          Demo
        </Button>
        <Button variant="secondary" isLoading>
          Demo
        </Button>
        <Button variant="outline" isLoading>
          Demo
        </Button>
      </Container>
      <h4>Disabled</h4>
      <Container>
        <Button disabled>Disabled</Button>
        <Button disabled variant="outline">
          Disabled
        </Button>
        <Button disabled variant="link">
          Disabled
        </Button>
        <Button disabled variant="primary" size="large" isLoading block>
          Disabled
        </Button>
      </Container>
      <h4>WithComponent</h4>
      <Container>
        <CustomComponent href="#somewhere" variant="primary">
          {'Button with <a href="#somewhere" /> tag'}
        </CustomComponent>
        <CloseButton />
      </Container>
      <h4>With Icon</h4>
      <Container>
        <Button icon={<Icon type="star" />}>With Icon</Button>
        <Button
          icon={<Icon type="starFilled" color={white} />}
          variant="primary"
        >
          With Icon
        </Button>
        <Button
          icon={<Icon type="starFilled" color={white} />}
          variant="secondary"
        >
          With Icon
        </Button>
        <Button
          icon={<Icon type="star" color={primary.main} />}
          variant="outline"
        >
          With Icon
        </Button>
      </Container>
    </div>
  ))
);
