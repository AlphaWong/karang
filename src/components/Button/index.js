import Button from './components/Button';
import CloseButton from './components/CloseButton';
import IconButton from './components/IconButton';
import LinkButton from './components/LinkButton';

export { Button as default, CloseButton, IconButton, LinkButton };
