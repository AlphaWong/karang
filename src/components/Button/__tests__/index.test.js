import React from 'react';
import { render } from 'enzyme';
import Button from '../index';

const types = {
  size: ['small', 'large', 'xlarge'],
  block: true,
  isLoading: true,
  variant: [
    'link',
    'primary',
    'secondary',
    'outline',
    'danger',
    'dangerOutline',
    'default',
  ],
  disabled: true,
};

for (const type in types) {
  if (type === 'variant' || type === 'size') {
    types[type].map(value => {
      it(`Button-${type}-${value}`, () => {
        const prop = { [type]: value };
        const wrapper = render(<Button {...prop}>hello</Button>);
        expect(wrapper).toMatchSnapshot();
      });
    });
  } else {
    it(`Button-${type}`, () => {
      const prop = { [type]: types[type] };
      const wrapper = render(<Button {...prop}>hello</Button>);
      expect(wrapper).toMatchSnapshot();
    });
  }
}
