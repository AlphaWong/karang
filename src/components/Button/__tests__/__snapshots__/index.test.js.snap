// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button-block 1`] = `
.c0 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: block;
  padding: 0.5em 1em;
  border: 1px currentColor solid;
  background-color: transparent;
  border-radius: 2px;
  font-family: "Noto Sans",sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1;
  text-align: center;
  cursor: pointer;
  outline: 0;
  display: block;
  width: 100%;
  border-color: rgba(229,229,229,1);
  color: rgba(88,89,91,1);
}

.c0:disabled {
  opacity: 0.3;
  cursor: not-allowed;
}

.c0:hover:enabled,
.c0:focus:enabled {
  box-shadow: 0 2px 4px rgba(0,0,0,0.22);
}

.c0:active:enabled {
  box-shadow: none;
}

<button
  class="c0"
  type="button"
>
  hello
</button>
`;

exports[`Button-disabled 1`] = `
.c0 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: block;
  padding: 0.5em 1em;
  border: 1px currentColor solid;
  background-color: transparent;
  border-radius: 2px;
  font-family: "Noto Sans",sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1;
  text-align: center;
  cursor: pointer;
  outline: 0;
  display: inline-block;
  border-color: rgba(229,229,229,1);
  color: rgba(88,89,91,1);
}

.c0:disabled {
  opacity: 0.3;
  cursor: not-allowed;
}

.c0:hover:enabled,
.c0:focus:enabled {
  box-shadow: 0 2px 4px rgba(0,0,0,0.22);
}

.c0:active:enabled {
  box-shadow: none;
}

<button
  class="c0"
  disabled=""
  type="button"
>
  hello
</button>
`;

exports[`Button-isLoading 1`] = `
.c0 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: block;
  padding: 0.5em 1em;
  border: 1px currentColor solid;
  background-color: transparent;
  border-radius: 2px;
  font-family: "Noto Sans",sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1;
  text-align: center;
  cursor: pointer;
  outline: 0;
  display: inline-block;
  border-color: rgba(229,229,229,1);
  color: rgba(88,89,91,1);
  background-image: url(loader.gif);
  background-position: center;
  background-size: 16px;
  background-repeat: no-repeat;
  color: rgba(0,0,0,0);
}

.c0:disabled {
  opacity: 0.3;
  cursor: not-allowed;
}

.c0:hover:enabled,
.c0:focus:enabled {
  box-shadow: 0 2px 4px rgba(0,0,0,0.22);
}

.c0:active:enabled {
  box-shadow: none;
}

<button
  class="c0"
  type="button"
>
  hello
</button>
`;

exports[`Button-size-large 1`] = `
.c0 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: block;
  padding: 0.5em 1em;
  border: 1px currentColor solid;
  background-color: transparent;
  border-radius: 2px;
  font-family: "Noto Sans",sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1;
  text-align: center;
  cursor: pointer;
  outline: 0;
  display: inline-block;
  padding: 0.75em 1em;
  font-size: 16px;
  font-weight: 700;
  border-color: rgba(229,229,229,1);
  color: rgba(88,89,91,1);
}

.c0:disabled {
  opacity: 0.3;
  cursor: not-allowed;
}

.c0:hover:enabled,
.c0:focus:enabled {
  box-shadow: 0 2px 4px rgba(0,0,0,0.22);
}

.c0:active:enabled {
  box-shadow: none;
}

<button
  class="c0"
  type="button"
>
  hello
</button>
`;

exports[`Button-size-xlarge 1`] = `
.c0 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: block;
  padding: 0.5em 1em;
  border: 1px currentColor solid;
  background-color: transparent;
  border-radius: 2px;
  font-family: "Noto Sans",sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1;
  text-align: center;
  cursor: pointer;
  outline: 0;
  display: inline-block;
  padding: 0.9em 1em;
  font-size: 18px;
  font-weight: 700;
  border-color: rgba(229,229,229,1);
  color: rgba(88,89,91,1);
}

.c0:disabled {
  opacity: 0.3;
  cursor: not-allowed;
}

.c0:hover:enabled,
.c0:focus:enabled {
  box-shadow: 0 2px 4px rgba(0,0,0,0.22);
}

.c0:active:enabled {
  box-shadow: none;
}

<button
  class="c0"
  type="button"
>
  hello
</button>
`;

exports[`Button-variant-default 1`] = `
.c0 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: block;
  padding: 0.5em 1em;
  border: 1px currentColor solid;
  background-color: transparent;
  border-radius: 2px;
  font-family: "Noto Sans",sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1;
  text-align: center;
  cursor: pointer;
  outline: 0;
  display: inline-block;
  border-color: rgba(229,229,229,1);
  color: rgba(88,89,91,1);
}

.c0:disabled {
  opacity: 0.3;
  cursor: not-allowed;
}

.c0:hover:enabled,
.c0:focus:enabled {
  box-shadow: 0 2px 4px rgba(0,0,0,0.22);
}

.c0:active:enabled {
  box-shadow: none;
}

<button
  class="c0"
  type="button"
>
  hello
</button>
`;

exports[`Button-variant-link 1`] = `
.c0 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: block;
  padding: 0.5em 1em;
  border: 1px currentColor solid;
  background-color: transparent;
  border-radius: 2px;
  font-family: "Noto Sans",sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1;
  text-align: center;
  cursor: pointer;
  outline: 0;
  display: inline-block;
  border: none;
  color: rgba(242,103,34,1);
}

.c0:disabled {
  opacity: 0.3;
  cursor: not-allowed;
}

.c0:hover:enabled,
.c0:focus:enabled {
  box-shadow: 0 2px 4px rgba(0,0,0,0.22);
}

.c0:active:enabled {
  box-shadow: none;
}

.c0:active {
  box-shadow: none;
  color: #a43d0a;
  -webkit-text-decoration: none !important;
  text-decoration: none !important;
}

.c0:hover:enabled,
.c0:focus:enabled {
  box-shadow: none;
  color: rgba(242,103,34,0.7);
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<button
  class="c0"
  type="button"
>
  hello
</button>
`;

exports[`Button-variant-outline 1`] = `
.c0 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: block;
  padding: 0.5em 1em;
  border: 1px currentColor solid;
  background-color: transparent;
  border-radius: 2px;
  font-family: "Noto Sans",sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1;
  text-align: center;
  cursor: pointer;
  outline: 0;
  display: inline-block;
  color: rgba(242,103,34,1);
  border-color: rgba(242,103,34,1);
}

.c0:disabled {
  opacity: 0.3;
  cursor: not-allowed;
}

.c0:hover:enabled,
.c0:focus:enabled {
  box-shadow: 0 2px 4px rgba(0,0,0,0.22);
}

.c0:active:enabled {
  box-shadow: none;
}

<button
  class="c0"
  type="button"
>
  hello
</button>
`;

exports[`Button-variant-primary 1`] = `
.c0 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: block;
  padding: 0.5em 1em;
  border: 1px currentColor solid;
  background-color: transparent;
  border-radius: 2px;
  font-family: "Noto Sans",sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1;
  text-align: center;
  cursor: pointer;
  outline: 0;
  display: inline-block;
  background-color: rgba(242,103,34,1);
  border-color: rgba(242,103,34,1);
  color: white;
}

.c0:disabled {
  opacity: 0.3;
  cursor: not-allowed;
}

.c0:hover:enabled,
.c0:focus:enabled {
  box-shadow: 0 2px 4px rgba(0,0,0,0.22);
}

.c0:active:enabled {
  box-shadow: none;
}

<button
  class="c0"
  type="button"
>
  hello
</button>
`;

exports[`Button-variant-secondary 1`] = `
.c0 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: block;
  padding: 0.5em 1em;
  border: 1px currentColor solid;
  background-color: transparent;
  border-radius: 2px;
  font-family: "Noto Sans",sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1;
  text-align: center;
  cursor: pointer;
  outline: 0;
  display: inline-block;
  background-color: rgba(250,162,27,1);
  border-color: rgba(250,162,27,1);
  color: white;
}

.c0:disabled {
  opacity: 0.3;
  cursor: not-allowed;
}

.c0:hover:enabled,
.c0:focus:enabled {
  box-shadow: 0 2px 4px rgba(0,0,0,0.22);
}

.c0:active:enabled {
  box-shadow: none;
}

<button
  class="c0"
  type="button"
>
  hello
</button>
`;
